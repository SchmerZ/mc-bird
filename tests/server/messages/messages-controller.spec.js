import Controller from '../../../src/server/api/messages/messages-controller'

const defaultConfig = {
  AccessKey: 'some-value',
  Originator: 'Inbox',
  urls: {
    MessageBirdApiRoot: 'http://rest.messagebird.com/api'
  }
};

const wsServer = {
  push: jest.fn(),
};

const defaultRequest = {body: {recipient: '79201010203', messageText: 'Hello!'}};

describe('Messages controller', () => {
  it('should send correct body on sendMessage call', async () => {
    let postData = null;
    const fn = jest.fn((args) => postData = args);

    const controller = new Controller(defaultRequest, null, null, defaultConfig, wsServer);
    controller.service.sendMessage = fn;

    await controller.sendMessage();

    expect(postData).toMatchSnapshot();
  });

  it('should push response message into WebSocket server', async () => {
    const responseData = {
      id: 'new-message-id',
      body: 'Hello back!',
    };

    let pushedMessage = null;
    wsServer.push = jest.fn(message => pushedMessage = message);

    const controller = new Controller(defaultRequest, null, null, defaultConfig, wsServer);
    controller.service.sendMessage = jest.fn().mockResolvedValueOnce(responseData);

    await controller.sendMessage();

    expect(pushedMessage).toEqual(responseData);
  });
});
