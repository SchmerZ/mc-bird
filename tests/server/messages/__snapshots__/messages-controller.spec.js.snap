// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages controller should push response message into WebSocket server 1`] = `
Object {
  "body": "Hello back!",
  "createdDatetime": undefined,
  "direction": undefined,
  "id": "new-message-id",
  "originator": undefined,
  "recipient": "79001007745",
  "status": "sent",
}
`;

exports[`Messages controller should send correct body on sendMessage call 1`] = `
Object {
  "body": "Hello!",
  "originator": "Inbox",
  "recipients": Array [
    "79201010203",
  ],
}
`;
